packages/core/saved-objects/core-saved-objects-base-server-internal/src/validation/validator.ts

```diff
@@ -56,10 +57,10 @@ export class SavedObjectsTypeValidator {
     }
     const schemaVersion = previousVersionWithSchema(this.orderedVersions, usedVersion);
 
-    if (!schemaVersion || !this.validationMap[schemaVersion]) {
-      return;
+    let validationRule: SavedObjectsValidationSpec | undefined;
+    if (schemaVersion && this.validationMap[schemaVersion]) {
+      validationRule = this.validationMap[schemaVersion];
```

Comment:
So, while adapting the validation, I discovered a somewhat big hole in the validation system: we were only performing it if an attribute schema was defined for the type/version tuple, which is a bug: we do want to validate the SO payload regardless.